lib/active_merchant/billing/gateways/orbital.rb

```diff
@@ -507,9 +507,33 @@ def add_managed_billing(xml, options)
       end
 
       def add_stored_credentials(xml, parameters)
-        xml.tag! :MITMsgType, parameters[:mit_msg_type] if parameters[:mit_msg_type]
-        xml.tag! :MITStoredCredentialInd, parameters[:mit_stored_credential_ind] if parameters[:mit_stored_credential_ind]
-        xml.tag! :MITSubmittedTransactionID, parameters[:mit_submitted_transaction_id] if parameters[:mit_submitted_transaction_id]
+        return unless parameters[:mit_stored_credential_ind] && parameters[:mit_stored_credential_ind] == 'Y' || parameters[:stored_credential] && !parameters[:stored_credential].values.all?(&:nil?)
+        if msg_type = get_msg_type(parameters)
+          xml.tag! :MITMsgType, msg_type
+        end
+        xml.tag! :MITStoredCredentialInd, 'Y'
+        if parameters[:mit_submitted_transaction_id]
+          xml.tag! :MITSubmittedTransactionID, parameters[:mit_submitted_transaction_id]
+        elsif parameters[:stored_credential] && parameters[:stored_credential][:network_transaction_id] && parameters[:stored_credential][:initiator] && parameters[:stored_credential][:initiator] == 'merchant'
```

Comment:
Likewise, can we simplify
`parameters[:stored_credential][:initiator] && parameters[:stored_credential][:initiator] == 'merchant'`
to
`parameters.dig(:stored_credential, :initiator) == 'merchant'` ?

Reply:
Nice, I think so; I didn't properly understand `.dig` when I went back and implemented it, and missed this refactor.